.github/workflows/prepare-release-branch.yml

```diff
@@ -0,0 +1,77 @@
+    steps:
+      - uses: actions/checkout@v2.3.4
+
+      - name: Set release branch name
+        id: set-release-branch-name
+        run: |
+          release_branch_name=$(grep -Eo "[0-9.]+-SNAPSHOT" version.gradle.kts | sed -E 's/([0-9]+)\.([0-9]+)\.([0-9]+)-SNAPSHOT/v\1.\2.x/')
+          echo "::set-output name=release-branch-name::$release_branch_name"
+
+      - name: Create release branch
+        run: |
+          git checkout -b ${{ steps.set-release-branch-name.outputs.release-branch-name }}
+          git push origin ${{ steps.set-release-branch-name.outputs.release-branch-name }}
+
+  create-pull-request-against-release-branch:
+    needs: prepare-release-branch
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ needs.prepare-release-branch.outputs.release-branch-name }}
+
+      - name: Remove SNAPSHOT from version
+        run: |
+          v=$(grep -Eo "[0-9]+.[0-9]+.[0-9]+-SNAPSHOT" version.gradle.kts | sed 's/-SNAPSHOT//')
+          .github/scripts/update-versions.sh "$v-SNAPSHOT" "$v-alpha-SNAPSHOT" "$v" "$v-alpha"
+
+      - name: Bump download link version
+        run: |
+          new_version=$(grep -Eo "[0-9]+.[0-9]+.[0-9]+" version.gradle.kts | head -1)
+          sed -Ei "s,https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v[0-9]+.[0-9]+.[0-9]+/,https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v$new_version/," README.md
+
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Prepare release branch ${{ needs.prepare-release-branch.outputs.release-branch-name }}
+          branch: prepare-release-branch-${{ needs.prepare-release-branch.outputs.release-branch-name }}
+          delete-branch: true
+          title: Prepare release branch ${{ needs.prepare-release-branch.outputs.release-branch-name }}
+          body: Prepare release branch ${{ needs.prepare-release-branch.outputs.release-branch-name }}
+
+  create-pull-request-against-main:
+    needs: prepare-release-branch
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+
+      - name: Bump SNAPSHOT version
+        run: |
+          v=$(grep -Eo "[0-9]+.[0-9]+.[0-9]+-SNAPSHOT" version.gradle.kts | sed 's/-SNAPSHOT//')
+          if [[ $v =~ ([0-9]+).([0-9]+).0 ]]; then
+            major="${BASH_REMATCH[1]}"
+            minor="${BASH_REMATCH[2]}"
+          else
+            echo "unexpected version: $v"
+            exit 1
+          fi
+          bump="$major.$((minor + 1)).0"
+          .github/scripts/update-versions.sh "$v-SNAPSHOT" "$v-alpha-SNAPSHOT" "$bump-SNAPSHOT" "$bump-alpha-SNAPSHOT"
+
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Bump snapshot version
+          branch: bump-snapshot-version
+          delete-branch: true
```

Comment:
FWIW, I was trying to see how delete-branch could possibly work when creating a pull request, and indeed it doesn't

https://github.com/peter-evans/create-pull-request#action-behaviour

It would delete the PR if it had already been merged when running this step. So to prevent confusion, we should delete this field if sticking with the action. But I'd say the CLI command is simple and all we need, this action seems to support some intense workflows :O

https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#how-the-action-works